### DISCUSSION TITLE
Displayed speed wrong

### DISCUSSION THREAD

**Discussion by turfutils** [gpx-animator/gpx-animator#679] [Q&A]
When I do an animation of a track, the speed is always around 0-1 km/h. If I set the Speedup to 45+ I start to get a believable speed closer to what RideWithGps says.

Thanks for a nice program!

**Comment by McPringle** [comment]
Does you recording device record your speed? GPX Animator uses the speed recorded in the GPX file. Because the video is usually not realtime and a video frame does not match a GPX point, GPX Animator has to "smooth out" the speed values and calculate an average corresponding to the video frame(s). A human would not be able to identify 30 or 60 different speed values per video second (videos usually play 30 or 60 frames per second), GPX Animator has to smooth out the values much more.

The default speedup value is 1'000. So 1'000 seconds of bike ride result in 1 second of video. If you stop at a stop sign for a few seconds or stay at a red light for a minute etc. that lowers your average speed for this one second of video. Technically, there is not much that I can do if you want to display your speed in the video. I'm happy if someone has a better calculation of the speed value to display in the video and appreciate any kind of contribution.

I have a RideWithGPS account, too, but could not find a feature to generate and export a video.

**Reply by turfutils** [reply]
No, there is no way of outputting videos in RideWithGPS, I'm just looking at points in the graph that displays the speed and compare with that. I don't see any speed in the gpx-files either,

I guess the only way to get speed out of the GPX data would be to calculate the distance between each point using the Haversine formula, add together 5-10 points and use the time difference of start-end to calculate speed.

https://stackoverflow.com/questions/3694380/calculating-distance-between-two-points-using-latitude-longitude

It has to be 25+ years since I touched Java, had a look at the code but can't find where to begin even :-)
